test/e2e/main_test.go

```diff
@@ -78,19 +78,50 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
+type InstallTestCase struct {
+	Name      string
+	ExpectErr bool
+	TestFunc  func(t *testing.T, testReportingFramework *reportingframework.ReportingFramework)
+}
+
 func TestInstallMeteringAndReportingProducesData(t *testing.T) {
 	testInstallConfigs := []struct {
 		TargetPods                int
 		Name                      string
 		MeteringOperatorImageRepo string
 		MeteringOperatorImageTag  string
 		MeteringConfigSpec        metering.MeteringConfigSpec
+		InstallSubTests           []InstallTestCase
 	}{
+		{
+			Name:                      "DefaultMeteringConfigProducesError",
+			TargetPods:                defaultTargetPods,
+			MeteringOperatorImageRepo: meteringOperatorImageRepo,
+			MeteringOperatorImageTag:  meteringOperatorImageTag,
+			InstallSubTests: []InstallTestCase{
+				{
+					Name:      "testReportingProducesData",
+					TestFunc:  testReportingProducesData,
+					ExpectErr: true,
+				},
+			},
+			MeteringConfigSpec: metering.MeteringConfigSpec{},
+		},
 		{
 			Name:                      "HDFSInstall",
 			TargetPods:                defaultTargetPods,
 			MeteringOperatorImageRepo: meteringOperatorImageRepo,
 			MeteringOperatorImageTag:  meteringOperatorImageTag,
+			InstallSubTests: []InstallTestCase{
+				{
+					Name:     "testReportingProducesData",
+					TestFunc: testReportingProducesData,
+				},
+				{
+					Name:     "testReportingProducesCorrectDataForInput",
```

Comment:
Im realizing this won't work exactly because `MeteringConfigSpec` needs to be different between these two cases, as `testReportingProducesCorrectDataForInput` should disable metrics importing for reporting-operator. I think the concept of `InstallSubTests` still makes sense, so we can leave it, but we need to move this to it's own parent test case with it's own meteringconfig.